src/common/sendpacket.c

```diff
@@ -1333,3 +1381,159 @@ sendpacket_abort(sendpacket_t *sp)
 
     sp->abort = true;
 }
+#ifdef HAVE_LIBXDP
+static struct xsk_socket_info *xsk_configure_socket(struct xsk_umem_info *umem, struct xsk_socket_config* cfg, int queue_id, const char *device)
+{
+	struct xsk_socket_info *xsk;
+	struct xsk_ring_cons *rxr = NULL;
+	struct xsk_ring_prod *txr;
+	int ret;
+	xsk = (struct xsk_socket_info*)safe_malloc(sizeof(struct xsk_socket_info));
+	xsk->umem = umem;
+	ret = xsk_socket__create(&xsk->xsk, device, queue_id, umem->umem, rxr, &xsk->tx, cfg);
+	if (ret){
+        return NULL;
+    }
+
+	xsk->app_stats.rx_empty_polls = 0;
+	xsk->app_stats.fill_fail_polls = 0;
+	xsk->app_stats.copy_tx_sendtos = 0;
+	xsk->app_stats.tx_wakeup_sendtos = 0;
+	xsk->app_stats.opt_polls = 0;
+	xsk->app_stats.prev_rx_empty_polls = 0;
+	xsk->app_stats.prev_fill_fail_polls = 0;
+	xsk->app_stats.prev_copy_tx_sendtos = 0;
+	xsk->app_stats.prev_tx_wakeup_sendtos = 0;
+	xsk->app_stats.prev_opt_polls = 0;
```

Comment:
memeset(&xsk->app_stats, 0, sizeof(xsk->app_stats) might also ok

Reply:
you are right